File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/validation/custom/HolidayService.kt
package io.micronaut.docs.validation.custom

import java.time.Duration
import jakarta.inject.Singleton
import jakarta.validation.constraints.NotBlank

// tag::class[]
@Singleton
open class HolidayService {

    open fun startHoliday(@NotBlank person: String,
                          @DurationPattern duration: String): String {
        val d = Duration.parse(duration)
        return "Person $person is off on holiday for ${d.toMinutes()} minutes"
    }
}
// end::class[]

File: test-suite-python/src/test/python/micronaut/docs/validation/custom/HolidayService.py
from typing import Annotated

from jakarta.inject import Singleton
from jakarta.validation.constraints import NotBlank
from java.time import Duration

from .DurationPattern import DurationPattern


# tag::class[]
@Singleton
class HolidayService:

    # tag::method[]
    def start_holiday(
        self,
        person: Annotated[str, NotBlank],
        duration: Annotated[str, DurationPattern],
    ) -> str:
        d = Duration.parse(duration)
        return f"Person {person} is off on holiday for {d.toMinutes()} minutes"
    # end::method[]
# end::class[]
